docs/UsersGuide.asciidoc

```diff
@@ -734,13 +738,91 @@ scenarios:
       testsuite: gnome
       machine: [32bit, 64bit-staging]
       settings:
-        FOO: 2
+        FOO: '2'
     opensuse-15.2-GNOME-Live-x86_64:
     - textmode
     - gnome: *gnome
     - gnome_staging: *gnome_staging
 ```
 
+=== YAML Merge Keys
+
+Also link:https://yaml.org/type/merge.html[YAML Merge Keys] are supported.
+This way you can reuse previously defined anchors and add other values to it.
+Values in the merged alias will be overriden.
+
+You can even merge more than one alias.
+
+
+```yaml
+products:
+  opensuse-15.1-DVD-Updates-x86_64:
+    distri: opensuse
+    flavor: DVD-Updates
+    version: '15.1'
+  opensuse-15.2-GNOME-Live-x86_64:
+    distri: opensuse
+    flavor: GNOME-Live
+    version: '15.2'
+scenarios:
+  x86_64:
+    opensuse-15.1-DVD-Updates-x86_64:
+    - textmode
+    - gnome:
+      machine: uefi
+      priorty: 70
+      settings: &common1
+        QEMUVGA: cirrus
+        FOO: default foo
+    - gnome:
+      machine: [32bit, 64bit-staging]
+      priorty: 70
+      settings: &common2
+        QEMUVGA: cirrus
+        FOO: default foo
+        BAR: default bar
+    - gnome_staging:
+      testsuite: gnome
+      machine: [32bit, 64bit-staging]
+      settings:
+        # Merge
+        << : *common1
+        FOO: foo # overrides the value from the merge keys
+    - gnome_staging:
+      testsuite: gnome
+      machine: [32bit, 64bit-staging]
+      settings:
+        # Merge
+        << : [*common1, *common2] # *common1 overrides *common2
+        FOO: foo # overrides the value from the merge keys
+```
+
+
+=== General YAML documentation
+
+The job templates are written in link:https://yaml.org/[YAML 1.2]. In YAML,
+strings usually do not have to be quoted, except if it is a special value that
+would be loaded as a Boolean, NULL or Number. The following table shows all
+regular expressions for special values.
+
+[%header,cols="70%,30%"]
+|=======
+| Regular expression                                                    | YAML Type
+| `true \| True \| TRUE \| false \| False \| FALSE`                     | `bool`
+| `0o [0-7]+`                                                           | `int (Base 8)`
+| `[-+]? [0-9]+`                                                        | `int (Base 10)`
+| `0x [0-9a-fA-F]+`                                                     | `int (Base 16)`
+| `[-+]? ( \. [0-9]+ \| [0-9]+ ( \. [0-9]* )? ) ( [eE] [-+]? [0-9]+ )?` | `float (Number)`
+| `[-+]? \. ( inf \| Inf \| INF )`                                      | `float (Infinity)`
+| `\. ( nan \| NaN \|NAN )`                                             | `float (Not a number)`
+| `null \| Null \| NULL \| ~`                                           | `null`
+| `^$`                                                                  | `null`
+| `.+`                                                                  | `str (Default)`
+|=======
```

Comment:
Maybe it's just me... but I feel like examples would be easier and more useful than regular expressions, from the perspective of someone using these types rather than implementing the parser :thinking:

Reply:
Ok. I think the true/false/null regexes are self explanatory, though.
I can add examples for the numbers.

Reply:
Yeah. Put it this way, if I'm reading this I'm probably wondering if `1.2f`, `false` or `15` should be quoted.

Reply:
Have to agree. Examples are better than regexes in this context.

Reply:
Ok, I now made it a table of examples, with additional regular expressions where necessary